quanta/server/module.config.js

```diff
@@ -0,0 +1,35 @@
+import mem from "mem";
+import configService from "server/core/config/configService";
+
+const memoizedMagentoExtraHeaders = mem(
+  (quantaHeader) => {
+    return {
+      magento: {
+        api: {
+          extraHeaders: {
+            "x-quanta": quantaHeader,
+          },
+        },
+      },
+    };
+  },
+  { maxAge: 5000 } // prevent memory leaks in case lots of different values pile up
+);
+
+const quantaConfigProvider = {
+  name: "quanta",
+  /**
+   * @param {import("express").Request} req
+   */
+  slowValuesOnEachRequest: (req) => {
+    const quantaHeader = req.header("x-quanta");
+    if (quantaHeader) {
+      return memoizedMagentoExtraHeaders(quantaHeader);
+    }
+    return {};
```

Comment:
Won't this trigger the warning about non memoized value coming from `slowValuesOnEachRequest` ?

Reply:
Totally! Good catch.

I amended the last commit to do it in the memoized function instead.
Thanks.